app/models/host_report.rb

```diff
@@ -55,4 +55,53 @@ def self.search_by_keyword(_key, operator, value)
       conditions: sanitize_sql_for_conditions(["host_reports.report_keyword_ids @> ?", "{#{keyword_ids.join(',')}}"]),
     }
   end
+
+  def self.report
+    @report = {} if @report.nil?
+    @report.update(
+      :total_hosts => Host.all.count,
+      :change_hosts => report_changes.count,
+      :nochange_hosts => report_nochanges.count,
+      :failure_hosts => report_failures.count,
+      :disabled_hosts => disabled_alerts.count,
+      :reports_missing => report_missing.count
+    )
+  end
+
+  def self.report_missing
+    Host.search_for('not has last_report and status.enabled = true')
+  end
+
+  def self.disabled_alerts
+    hosts.where(:enabled => true)
+  end
+
+  def self.report_changes
+    hosts.search_for("report_changes = 1")
+  end
+
+  def self.report_nochanges
+    hosts.search_for("report_nochanges = 1")
+  end
+
+  def self.report_failures
+    hosts.search_for("report_failures = 1")
+  end
+
+  def self.latest_events?
+    latest_events.limit(1).present?
+  end
+
+  def self.hosts
+    filter = nil # TODO: implement global filtering
+    hosts || Host.authorized(:view_hosts, Host).search_for(filter)
+  end
+
+  def self.latest_events
+    @latest_events ||= HostReport.authorized(:view_host_reports).my_reports
+                                 .where(host_id: hosts.pluck(:id))
+                                 .reorder(:last_report => :desc)
+                                 .limit(9)
+                                 .preload(:host)
+  end
```

Comment:
Please rename, let's kill the old term "event".

Reply:
lateset_reports are OK?

Reply:
Yup that would work for me.